scootapi/server/runJob.go

```diff
@@ -0,0 +1,93 @@
+package server
+
+import (
+	"fmt"
+	"github.com/nu7hatch/gouuid"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/scheduler"
+	"github.com/scootdev/scoot/scootapi/gen-go/scoot"
+)
+
+// Implementation of the RunJob API
+func runJob(scheduler scheduler.Scheduler, def *scoot.JobDefinition) (*scoot.JobId, error) {
+
+	jobDef, err := thriftJobToScoot(def)
+	// TODO: change to return scoot.NewInvalidRequest()
+	if err != nil {
+		return nil, err
+	}
+
+	err = validateJob(jobDef)
+	if err != nil {
+		return nil, err
+	}
+
+	job := sched.Job{
+		Id:  generateJobId(),
+		Def: jobDef,
+	}
+
+	err = scheduler.ScheduleJob(job)
```

Comment:
ok yes thought about this! I will move this into scheduler.  Will submit in subsequent code review since it'll involve modifying a bunch of files & the Scheduler interface